### PR TITLE
wallet-ext: copy to clipboard use toast for confirmation feedback

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* show a toast instead of scaling the copy icon
* move copy to functionality to hooks to reuse it
* also removes unused code for rendering sui objects

before

https://user-images.githubusercontent.com/10210143/214037484-97443686-66eb-49bc-a550-fe39c7d6aa69.mov

after

https://user-images.githubusercontent.com/10210143/214036853-6cf71f8e-7b8e-4dda-acf5-c4c64d5fd3f9.mov

https://user-images.githubusercontent.com/10210143/214036872-53ca8228-75bc-4135-aa1a-71e643e92252.mov

part of APPS-296

### CODE REVIEW

File: apps/wallet/src/ui/app/hooks/useCopyToClipboard.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import { type MouseEventHandler, useCallback } from 'react';
+import { toast } from 'react-hot-toast';
+
+export type CopyOptions = {
+    copySuccessMessage?: string;
+};
+
+export function useCopyToClipboard(
+    txpToCopy: string,
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/hooks/useCopyToClipboard.ts:12]: `txtToCopy` ?

**Comment by mystie711** [comment]
Is the toast height correct? 
Feels a tad bit taller?  🤔

File: apps/wallet/src/ui/app/hooks/useCopyToClipboard.ts (hunk 1 above)
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/hooks/useCopyToClipboard.ts]: Thanks renamed it

**Comment by pchrysochoidis** [comment]
> Is the toast height correct? Feels a tad bit taller? 🤔

Should be the same as it already is in main, I did't change it.

**Comment by mystie711** [comment]
> > Is the toast height correct? Feels a tad bit taller? 🤔
> 
> Should be the same as it already is in main, I did't change it.

Could you please double check if we are using the correct size in main?

**Comment by pchrysochoidis** [comment]
> > > Is the toast height correct? Feels a tad bit taller? 🤔
> > 
> > 
> > Should be the same as it already is in main, I did't change it.
> 
> Could you please double check if we are using the correct size in main?

There was some extra margin from the lib, address [here](https://github.com/MystenLabs/sui/pull/7729)

**Review by Jibz1 — APPROVED** [review]

File: apps/wallet/src/ui/app/components/copy-to-clipboard/index.tsx [hunk 2]
```diff
@@ -26,42 +25,17 @@ function CopyToClipboard({
...
+    const copyToClipboard = useCopyToClipboard(txt, { copySuccessMessage });
     return (
         <span
             className={cl(st.container, className)}
             onClick={!copyOnlyOnIconClick ? copyToClipboard : undefined}
         >
             {children}
             <Icon
-                className={cl(st.copyIcon, st[mode], { [st.copied]: copied })}
+                className={cl(st.copyIcon, st[mode])}
```
Comment (Jibz1) [inline apps/wallet/src/ui/app/components/copy-to-clipboard/index.tsx:38]: can we use Tailwind here

**Review by Jibz1 — APPROVED** [review]

File: apps/wallet/src/ui/app/components/copy-to-clipboard/index.tsx (hunk 2 above)
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/components/copy-to-clipboard/index.tsx:38]: We can but this component has other usages of css modules and classes and I think it's better to address them in an another PR that will do only that

**Pull request merged**